packages/ui/src/SelectField.tsx

```diff
@@ -31,6 +31,53 @@ const Input = (props: any) => {
   return <components.Input {...props} autoComplete="chrome-off" />
 }
 
+// This component is implemented because we want to style things with the css (as opposed to the react-select
+// recommended way of styling things via js).
+//
+// We simply inject `.multiValueIsFocused`.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const MultiValue = (props: MultiValueProps<any>) => {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  props.components.Remove = MultiValueRemove
+
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  props.components.Label = MultiValueLabel
+
+  return <components.MultiValue {...props} className={cn(styles.multiValue, { [styles.multiValueIsFocused]: props.isFocused })} />
+}
+
+// Self styled version of the MultiValue/Label
+//
+// refactor: newer react-select packages have better typings for this.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const MultiValueLabel = (props: any) => {
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access,  @typescript-eslint/no-unsafe-assignment
+  const children = props.children
+  return <div className={cn(styles.multiValueLabel)}> {children} </div>
+}
+
+// Self styled version of the MultiValue/Remove button
+//
+// refactor: newer react-select packages have better typings for this.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const MultiValueRemove = (props: any) => {
+  // We have to pass down the onClick
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access,  @typescript-eslint/no-unsafe-assignment
+  const onClick: React.MouseEventHandler = props.innerProps.onClick
+
+  const handleMouseDown = (event: React.MouseEvent) => {
+    event.preventDefault()
+    event.stopPropagation()
+    event.button === 0 && onClick && onClick(event)
```

Comment:
@utkukaratas why do we need this line?

Reply:
Can't we just pass `onClick` to the button? BTW, should we just use `IconButton`?

Reply:
Oh, I see now that it is custom styled.The question regarding `onClick` still stands.

Reply:
@utkukaratas there is a small visual problem 
![image](https://user-images.githubusercontent.com/12794628/108244707-88272d00-7160-11eb-854e-dfef5f0c0ba8.png)

Reply:
As you can see the hover background for the remove button overlaps with the focus shadow. Shouldn't it go underneath?

Reply:
We'll probably need a transparent absolutely positioned element on top of the button to draw the shadow on the very top

Reply:
> @utkukaratas why do we need this line?

we need to pass the onclick given from the parent to our custom div. it's done in `mousedown`  to do everything in one fell swoop and keep the linter happy.

Reply:
@utkukaratas personally, I would expect the removal to happen only when I release the mouse, but it is a nit. It is just that click is usually happens after mouse is up, so currently it feels a bit unnatural. WDYT? What error does a linter give you?

Reply:
I tried it. It is a keybord listener error, right? I would suggest to just put a eslint ignore on this line.  WDYT?

Reply:
> We'll probably need a transparent absolutely positioned element on top of the button to draw the shadow on the very top

just modified the box shadow to be NOT inset ;)